Replace a few more uses of virtualenv with venv

yarikoptic opened the pull request.

christian-monch commented on .appveyor.yml:
```diff
@@ -238,9 +238,9 @@ install:
+  # Otherwise create a new venv using the default Python 3, to enable uniform
   # use of python/pip executables below
-  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || virtualenv -p 3 ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
+  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || python -m venv -p 3 ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
```
Should it not be `python3`? `python` resolves to a Python interpreter with version 2.7.18 in an ssh session in `Ubuntu2004`.  

```suggestion
  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || python3 -m venv -p 3 ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
```

yarikoptic replied:
yes! thanks for spotting @christian-monch !  I misread the code and assumed that we would be after effects of `. ${HOME}/venv${PY}/bin/activate` when we would be having that python to be python3.  So indeed some old systems could have python2 still be installed  so let's ensure that we have python3, but then `-p 3` is not needed and I think is not supported by venv since we execute it via `python3` itself anyways hence

```suggestion
  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || python3 -m venv ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
```

christian-monch reviewed (changes requested):
Thanks for the patches. Left a comment regarding python versions

mih reviewed (commented):
Just a brief comment: I think it would be more sensible to have this PR against the extension template. The load on the CI would be less, and it will need to be done there anyways.

yarikoptic commented:
worth doing there too but I guess it is not to be merged here anyways:

```
❯ git merge gh-template/main
Auto-merging .appveyor.yml
CONFLICT (content): Merge conflict in .appveyor.yml
Auto-merging .codespellrc
CONFLICT (add/add): Merge conflict in .codespellrc
Auto-merging .github/workflows/docbuild.yml
CONFLICT (content): Merge conflict in .github/workflows/docbuild.yml
Auto-merging .github/workflows/test_crippledfs.yml
CONFLICT (content): Merge conflict in .github/workflows/test_crippledfs.yml
Auto-merging CONTRIBUTING.md
CONFLICT (add/add): Merge conflict in CONTRIBUTING.md
Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
Auto-merging changelog.d/scriv.ini
CONFLICT (add/add): Merge conflict in changelog.d/scriv.ini
Auto-merging changelog.d/templates/new_fragment.md.j2
CONFLICT (add/add): Merge conflict in changelog.d/templates/new_fragment.md.j2
CONFLICT (modify/delete): datalad_helloworld/__init__.py deleted in HEAD and modified in gh-template/main.  Version gh-template/main of datalad_helloworld/__init__.py left in tree.
CONFLICT (modify/delete): datalad_helloworld/hello_cmd.py deleted in HEAD and modified in gh-template/main.  Version gh-template/main of datalad_helloworld/hello_cmd.py left in tree.
Auto-merging datalad_next/_version.py
Auto-merging docs/source/conf.py
CONFLICT (content): Merge conflict in docs/source/conf.py
Auto-merging readthedocs.yml
CONFLICT (add/add): Merge conflict in readthedocs.yml
Auto-merging setup.cfg
CONFLICT (content): Merge conflict in setup.cfg
Automatic merge failed; fix conflicts and then commit the result.
```

mih reviewed (commented):
I appreciate the enthusiasm. But all the energy here is spent on fixing something that is not broken.

I do pull in changes from the template.
